### DISCUSSION TITLE
Using Raw Input with binary OpenTabletDriver locks cursor to half the window

### DISCUSSION THREAD

**Discussion by slopslug** [ppy/osu#27823] [Ideas] [answered]
### Type

Game behaviour

### Bug description

On stable, using Raw Input works fine and causes no issues.
On lazer, in fullscreen, using the binary version of OpenTabletDriver causes the mouse to be locked to the right half of the screen.

When running lazer in windowed mode at a size larger than my display, it jitters between being inside of the window and outside.

Running OpenTabletDriver version `0.6.4.0`

### Screenshots or videos

https://github.com/ppy/osu/assets/47635446/24275129-1a22-4574-873c-4864f806114e

https://github.com/ppy/osu/assets/47635446/fcc0a74e-5b5a-4d0d-b9ff-85f6e41b9e56

### Version

2024.302.1-lazer

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14923087/compressed-logs.zip)

**Comment by Stoppedpuma** [comment, marked as answer]
Duplicate of many, first results within a quick search:

#24629
#26922

Solution is to either disable high precision mouse or use the built in driver.

**Comment by slopslug** [comment]
I commented on the first issue's discussion page and have not received a response since.

[cc](https://github.com/ppy/osu/discussions/24630#discussioncomment-6842517)

> Are there going to be any patches to allow us to use third party installations of OpenTabletDriver? I appreciate the thought of putting first party support into lazer, however without any particular way to migrate my configuration or the ability to use custom filters I can't see myself using the built-in version.

**Comment by Stoppedpuma** [comment]
> You have to understand that if you're using an external tablet driver, then the only way lazer can see this input is in the form of mouse events (as in, the tablet driver is emulating your tablet input into mouse input for the entire operating system). Stripping the name "Mouse" from the option is pretty much moving the issue from tablet users to mouse users, I do not see that as a valid solution.

Smoogis comment on the second post should explain this to you. This also happens in stable to a lot of people and has been general advice for years to disable raw input if it gives you problems (applies on lazer as well). Enabling raw input should only be a concern if there's acceleration to bypass, when it comes to absolute input from tablets there shouldn't be any additional acceleration.

**Reply by slopslug** [reply]
In that case, are there any plans to support migration to the built-in driver?

**Reply by Stoppedpuma** [reply]
Config imports, [probably not](https://github.com/ppy/osu/discussions/25134). Filters are a [maybe](https://github.com/ppy/osu/discussions/15306).

Search around in discussions and issues for questions related to tablet, you'll most likely find what you're looking for.
